### DISCUSSION TITLE
Application for nematodes?

### DISCUSSION THREAD

**Discussion by aardvark0420** [Abe404/root_painter#116] [Q&A] [answered]
Hi. To what extent is RootPainter applicable for image analysis of nematodes (from a  microscope)? If it has use, what features could I expect the tool to be trained on and recognize? Is it purely a tool for feature extraction (counting, subject tracking, etc.) or could it theoretically also be used to automate subject classification/identification, for example identifying genera.

**Comment by Abe404** [comment, marked as answer]
Probably it would work very well, if you can see the nematodes clearly in your images and detect them by eye. 

There are experimental 'multi class' version of RootPainter but in general a given model is trained to produce a binary map (segmentation) that corresponds to foreground and background in the image. It is up to you to decide what is foreground and background in your image (this is where you have a lot of flexibility). RootPainter has counting functions built in also.

Regarding subject tracking, this is not something I have developed or tested yet. 

If you can clearly see the difference between different nematode genera then you could train a model to segment only the (one or more) genera you are interested in.

Training models can be quite fast (depending on dataset, approach to annotation and hardware) so it's possible to train different models to analyse the same images/dataset in different ways. i.e you could train a model for each genera. 

Kind regards,
Abraham

**Reply by aardvark0420** [reply]
Thank you, Abraham, for your prompt response and explanation.
